sys/include/registry/string_path.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2023 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_registry_string_path RIOT Registry String Path
+ * @ingroup     sys
+ * @brief       RIOT Registry String Path module providing functions to convert between registry objects and their string paths
+ * @{
+ *
+ * @file
+ *
+ * @author      Lasse Rosenow <lasse.rosenow@haw-hamburg.de>
+ */
+
+#ifndef REGISTRY_STRING_PATH_H
+#define REGISTRY_STRING_PATH_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    REGISTRY_STRING_PATH_TYPE_NAMESPACE,
+    REGISTRY_STRING_PATH_TYPE_SCHEMA,
+    REGISTRY_STRING_PATH_TYPE_INSTANCE,
+    REGISTRY_STRING_PATH_TYPE_GROUP,
+    REGISTRY_STRING_PATH_TYPE_PARAMETER,
+    REGISTRY_STRING_PATH_TYPE_GROUP_OR_PARAMETER,
+}registry_string_path_type_t;
+
+#include "registry.h"
+
+/**
+ * @brief Converts a registry namespace object to its string path representation.
+ *
+ * @param[in] namespace The registry namespace object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_namespace_string_path(const registry_namespace_t *namespace, char *path);
+
+/**
+ * @brief Converts a registry schema object to its string path representation.
+ *
+ * @param[in] schema The registry schema object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_schema_string_path(const registry_schema_t *schema, char *path);
```

Comment:
I agree with moving the `_to_`, but I would prefer not to drop the string.
This would make the naming scheme inconsistent and less precise from my view.

So maybe: `registry_schema_to_string_path`
Or maybe make it more clear: `registry_convert_schema_to_string_path` (even though this is more long)

But if we choose to clean up all those many convert functions and move to a single function with a unified type, we could also have something like `registry_convert_to_string_path` or `registry_to_string_path`.

Reply:
I have opted for:

`registry_node_to_string_path`
and
`registry_node_from_string_path`

Reply:
This is also implemented already.